pkg/kubectl/cmd/util/shortcut_restmapper.go

```diff
@@ -40,13 +42,20 @@ func NewShortcutExpander(delegate meta.RESTMapper, client discovery.DiscoveryInt
 	return ShortcutExpander{All: userResources, RESTMapper: delegate, discoveryClient: client}
 }
 
-func (e ShortcutExpander) getAll() []schema.GroupResource {
+func (e ShortcutExpander) getServerRes(groupVersion string) ([]*metav1.APIResourceList, error) {
```

Comment:
I'd like to more narrowly scope this return value.  I think we need an order list of (shortname,group) tuples to (resource,group) tuples.  The ordering matters for priority.